tests/conformance/nonconforming.yaml

```diff
@@ -6,3 +6,11 @@ standard_constraints/well_known_types/duration:
 standard_constraints/well_known_types/timestamp:
  - gte_lte/invalid/above
  - lte/invalid
+# There are a couple of cases where celpy's mapping of has(...) and x.y seem to
+# be incongruent with protobuf presence, resulting in a runtime error instead of
+# a validation error.
```

Comment:
Can we report this upstream?

Reply:
I really need to do more investigation into the problem in order to understand what's going on, but I haven't had time. It's complicated slightly by the fact that we don't even use celpy's protobuf integration afaict, we just use MapTypes manually. I tried to use celpy's MessageType just to see if that would save us some trouble, but it had even more issues, so I think there's good reason for what we did.

Reply:
It really feels tricky to debug this but after probing the Go implementation and carefully reading the cel-python source code I think I have a good enough understanding of the problem. In order to confirm my understanding, I implemented a workaround in ab874b27d746fd5f1ab3b26f820a2f358ded42c0.

The workaround trivially passes our conformance tests and while the mechanism used to implement it is ugly, the actual logic is fairly simple and straight-forward. I committed the workaround thunks as their own commit so that we can drop this commit if we decide it's too ugly to include in protovalidate-python, but I hope by actually producing the workaround we can have a bit more confidence that we actually do understand the nature of the problem correctly. It's possible there are other issues lurking in the handling of presence and defaults (I think the conformance test suite needs to grow a lot...) but at the very least I think this covers the issue we're currently facing correctly and completely.

I reported the issue upstream, albeit without a minimal reproduction for now. I think it will at least need a minimal reproduction, but I already spent a lot more time on this issue than intended, so I might need to loop back to it after continuing onto other protovalidate implementations.

https://github.com/cloud-custodian/cel-python/issues/73

Reply:
I don't mind the patch, assuming it will continue to work for versions of cel-python that fix this particular code. Should we pin pv-python to be more tightly tied to these versions of cel-python?

Reply:
Yeah, I think we definitely should do that.

I'm hopeful that it can be fixed upstream, but it may be a "PRs welcome" type of situation, and if so I'll need to try picking up some lark-parser knowledge.